### ISSUE TITLE
Bug: user sessions are never properly cleaned up when a node dies 

### ISSUE THREAD

**Issue by elliottshort** [copleykj/socialize-user-presence#25]
Perhaps I'm overlooking something, but it seems that this query should be `{ serverId: serverId }` instead, yes? 
Thanks! 

https://github.com/copleykj/socialize-user-presence/blob/8e8d6fe509f7e339940a5368aae03ca167cff0df/server/user-presence.js#L148

**Comment by copleykj** [comment]
No, this is just short hand property syntax and is exactly the same functionally. 

Can you give further details about the issue?

**Comment by elliottshort** [comment]
Hey @copleykj – thanks for the prompt reply! I did some additional troubleshooting and it seems that the varying issues I'm experiencing are likely related to the fact I'm using https://github.com/cult-of-coders/redis-oplog instead of Meteor's built-in. I'll likely have to wind up rolling something of my own unfortunately. 

I'll go ahead and close this issue. Cheers!

**Comment by copleykj** [comment]
I'm gonna reopen this cause the rest of the packages in this set are built to work with redis-oplog and if something about it breaks this package I want to understand what might cause this and fix it.

**Comment by elliottshort** [comment]
Sure thing. The key issue seems to be that that redis oplog doesn't have any means of detecting the db TTL/record expiry that the server presence package relies on: `Servers.createIndex({ lastPing: 1 }, { expireAfterSeconds: 10 });` 

The workaround I've implemented is to use the https://www.npmjs.com/package/mongo-oplog2/v/2.1.2 package and tail the oplog myself -> push the changes to redis so that the logic in the server presence package associated with `Servers.find().observe(...` works properly. 

Workaround: 
```
const oplog = new MongoOplog(
  Meteor.settings["galaxy.meteor.com"].env.MONGO_OPLOG_URL,
  { ns: "meteor.presence:servers" }
);

oplog.tail().then();

oplog.on("delete", (document: OplogDoc) => {
  getRedisPusher().publish(
    "presence:servers",
    EJSON.stringify({
      [RedisPipe.DOC]: { _id: document.o._id },
      [RedisPipe.EVENT]: Events.REMOVE,
      [RedisPipe.FIELDS]: ["createdAt", "lastPing", "watcher"],
    })
  );
});
```

Hope that helps!
